src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Linux.cs

```diff
@@ -33,11 +33,13 @@ public static string[] GetPortNames()
                     // checking for that for non-ttyS entries is incorrect as some uart
                     // devices are incorrectly filtered out
                     bool isTtyS = entry.Name.StartsWith("ttyS");
+                    bool isTtyGS = entry.Name.StartsWith("ttyGS");
```

Comment:
isTtyGS can't be true if isTtyS is true, as they're checking for competing conditions.
```suggestion
                    bool isTtyGS = !isTtyS && entry.Name.StartsWith("ttyGS");
```